ipn/ipnlocal/profiles.go

```diff
@@ -544,7 +546,7 @@ func newProfileManagerWithGOOS(store ipn.StateStore, logf logger.Logf, goos stri
 		if err := pm.setPrefsLocked(prefs); err != nil {
 			return nil, err
 		}
-	} else if len(knownProfiles) == 0 && goos != "windows" {
+	} else if len(knownProfiles) == 0 && goos != "windows" && runtime.GOOS != "windows" {
```

Comment:
I don't follow why we need to check both of these. If that's intentional (which I assume it is) a comment might be worthwhile.

Reply:
It is intentional, I'll add a comment. I think this needs some extra cleanup - I think the intent was for most behavior to be driven by goos parameter, but that isn't consistently the case, and we execute tests with `goos == 'linux'` in the Windows pipeline which implies behavior by build tag which otherwise fails.